spec/factories/course_assessment_submissions.rb

```diff
@@ -6,8 +6,14 @@
     transient do
       grader { User.stamper }
       auto_grade true # Used only with any of the submitted or finalised traits.
+      user { creator }
     end
     assessment { build(:assessment, course: course) }
+    creator
+
+    after(:build) do |submission, evaluator|
+      submission.creator = evaluator.user
```

Comment:
I do believe that the userstamp gem does not override a creator if it is already there, if that's not the case, a PR is welcome :)